### PR TITLE
DOMA-2636 ticket comments type backend

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**

### CODE REVIEW

**Review by sitozzz — APPROVED** [review]

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/ticket/access/TicketComment.js [hunk 1]
```diff
@@ -32,21 +60,64 @@ async function canManageTicketComments ({ authentication: { item: user }, origin
...
+        const residents = await find('Resident', {
+            user: { id: user.id },
+            property: { id: propertyId, deletedAt: null },
+            unitName,
+            unitType,
+            deletedAt: null,
+        })
+
+        return residents.length > 0
+    } else if (user.type === STAFF) {
```
Comment (pahaz) [inline apps/condo/domains/ticket/access/TicketComment.js:104]: Use else. We also have a service accounts

File: apps/condo/domains/ticket/access/TicketComment.js [hunk 2]
```diff
@@ -60,6 +131,14 @@ async function canSetUserField ({ authentication: { item: user }, originalInput
     return get(originalInput, ['user', 'connect', 'id']) === user.id
 }
 
+async function canReadUserField ({ authentication: { item: user }, existingItem }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin || user.type === STAFF) return true
```
Comment (pahaz) [inline apps/condo/domains/ticket/access/TicketComment.js:137]: user.type !== Redisent

Comment (pahaz) [reply apps/condo/domains/ticket/access/TicketComment.js:137]: We also have a service user.type

File: apps/condo/migrations/20220411124706-0119_ticketcomment_type_ticketcommenthistoryrecord_type.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// auto generated by kmigrator
+// KMIGRATOR:0119_ticketcomment_type_ticketcommenthistoryrecord_type:IyBHZW5lcmF0ZWQgYnkgRGphbmdvIDQuMCBvbiAyMDIyLTA0LTExIDA3OjQ3Cgpmcm9tIGRqYW5nby5kYiBpbXBvcnQgbWlncmF0aW9ucywgbW9kZWxzCgoKY2xhc3MgTWlncmF0aW9uKG1pZ3JhdGlvbnMuTWlncmF0aW9uKToKCiAgICBkZXBlbmRlbmNpZXMgPSBbCiAgICAgICAgKCdfZGphbmdvX3NjaGVtYScsICcwMTE4X3JlbmFtZV9kZXRhaWxzdGV4dF9hY3F1aXJpbmdpbnRlZ3JhdGlvbl9pbnN0cnVjdGlvbl9hbmRfbW9yZScpLAogICAgXQoKICAgIG9wZXJhdGlvbnMgPSBbCiAgICAgICAgbWlncmF0aW9ucy5BZGRGaWVsZCgKICAgICAgICAgICAgbW9kZWxfbmFtZT0ndGlja2V0Y29tbWVudCcsCiAgICAgICAgICAgIG5hbWU9J3R5cGUnLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuVGV4dEZpZWxkKGRlZmF1bHQ9J29yZ2FuaXphdGlvbicpLAogICAgICAgICAgICBwcmVzZXJ2ZV9kZWZhdWx0PUZhbHNlLAogICAgICAgICksCiAgICAgICAgbWlncmF0aW9ucy5BZGRGaWVsZCgKICAgICAgICAgICAgbW9kZWxfbmFtZT0ndGlja2V0Y29tbWVudGhpc3RvcnlyZWNvcmQnLAogICAgICAgICAgICBuYW1lPSd0eXBlJywKICAgICAgICAgICAgZmllbGQ9bW9kZWxzLlRleHRGaWVsZChibGFuaz1UcnVlLCBudWxsPVRydWUpLAogICAgICAgICksCiAgICBdCg==
+
+exports.up = async (knex) => {
+    await knex.raw(`
+    BEGIN;
+--
+-- Add field type to ticketcomment
+--
+ALTER TABLE "TicketComment" ADD COLUMN "type" text DEFAULT 'organization' NOT NULL;
```
Comment (pahaz) [inline apps/condo/migrations/20220411124706-0119_ticketcomment_type_ticketcommenthistoryrecord_type.js:10]: You need to set the default value.

File: apps/condo/domains/ticket/access/TicketComment.js (hunk 1 above)
Comment (pahaz) [inline apps/condo/domains/ticket/access/TicketComment.js]: Type service should work as staff

**Pull request merged**